django/db/models/expressions.py

```diff
@@ -927,6 +927,89 @@ def convert_value(self, value, expression, connection, context):
         return value
 
 
+class SubQuery(Expression):
+    """
+    Notes about both SubQuery() and Exists().
+
+    MyModel.objects.annotate(
+        field=SubQuery(RelatedModel.objects.filter(date=Today())
+                                           .values('other_field')[:1]),
+        found=Exists(OtherModel.objects.filter(thing=True))
+    )
+
+    Note that for SubQuery, unless you are aggregating the sub query, you'll want to use
+    a [:1] on the subquery to use a LIMIT 1. The Exists() is not subject to this. You'll probably
+    also want to use a .values() with a single argument, unless you are using something that
+    will deal with multiple fields (like building a JSON object, maybe).
+
+    TODO: Multiple join fields.
+    TODO: Tests/example usage.
+    TODO: .filter(Exists()) support. Maybe .filter(field__where_exists=Exists()) ?
+    TODO: Auto-set output_field based on the subquery?
+
+    You can currently do it by .annotate(foo=Exists()).filter(foo=True)
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery
+        self.extra = extra
+        super(SubQuery, self).__init__(output_field or self.output_field)
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        # Update the subquery based on the query, which we now have a reference to.
+        clone = self.copy()
+        clone.is_summary = summarize
+
+        # We want to work on a clone of the subquery too. Since we will be mutating it.
+        subquery = clone.subquery.all()
+        subquery.query.where.children.append(self.build_subquery_filter(query, subquery))
+        clone.subquery = subquery
+        return clone
+
+    def build_subquery_filter(self, query, subquery):
+        # Circular reference if we do it earlier.
+        from django.db.models.lookups import Exact
+
+        # Is there a single relationship between subquery and query?
```

Comment:
Yes, the magic here is probably the thing I like least (although the `where.children.append()` also feels nasty, but maybe there isn't another way to do that).

I played around with different ways of passing stuff around (but that was before I was pointed towards `resolve_expression`)